src/test/ui/infinite/infinite-tag-type-recursion.stderr

```diff
@@ -8,6 +8,16 @@ LL | enum MList { Cons(isize, MList), Nil }
    |
    = help: insert indirection (e.g., a `Box`, `Rc`, or `&`) at some point to make `MList` representable
 
-error: aborting due to previous error
+error[E0391]: cycle detected when processing `MList`
+  --> $DIR/infinite-tag-type-recursion.rs:1:1
+   |
+LL | enum MList { Cons(isize, MList), Nil }
+   | ^^^^^^^^^^
+   |
+   = note: ...which again requires processing `MList`, completing the cycle
+   = note: cycle used when computing dropck types for `Canonical { max_universe: U0, variables: [], value: ParamEnvAnd { param_env: ParamEnv { caller_bounds: [], reveal: UserFacing, def_id: None }, value: MList } }`
```

Comment:
This note doesn't seem like something the user would understand... :)

Reply:
It doesn't. There hard error removal is causing some spurious and duplicated errors that I want to hide, but for now I want to put these changes out asap so that's we can capture any lurking ICEs.

Reply:
we have a lot of cycle errors without comprehensible diagnostics. This is something that needs to be addressed in general and is not specific to this PR

Reply:
Filed https://github.com/rust-lang/rust/issues/60178